### PR TITLE
Move code to exit servo after writing a screenshot out of composite_specific_target.

### PR DESCRIPTION
**Pull request by jgraham**
The structure of this function was confusing, so move some parts out into the
caller where they seem like a more natural fit and add documentation of the
functions

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7875)

### CODE REVIEW

**Comment by jgraham** [comment]
@glennw does something like this seem sensible? I didn't fully test it yet…

File: components/compositing/compositor.rs [hunk 1]
```diff
@@ -478,7 +478,8 @@ impl<Window: WindowMethods> IOCompositor<Window> {
             }
 
             (Msg::CreatePng(reply), ShutdownState::NotShuttingDown) => {
-                let img = self.composite_specific_target(CompositeTarget::WindowAndPng);
+                let res = self.composite_specific_target(CompositeTarget::WindowAndPng);
+                let img = res.ok().and_then(|x| x);
```
Comment (jdm) [inline components/compositing/compositor.rs]: The and_then seems redundant.

Comment (jgraham) [inline components/compositing/compositor.rs]: It unwraps a level of Option i.e. `Some(Some(img))` becomes `Some(img)`

Comment (jgraham) [inline components/compositing/compositor.rs]: But I guess I'm looking for `unwrap_or`. The `Result` and `Option` APIs are huge :(

**Comment by glennw** [comment]
@jgraham It looks reasonable to me - did you want to do any more testing / changes on it, or is it ready?

**Comment by jgraham** [comment]
I think it's ready for review.

**Comment by jdm** [comment]
@bors-servo: retry
* #7730

**Comment by jdm** [comment]
@bors-servo: retry
* #7730

**Comment by jdm** [comment]
@bors-servo: retry
* #7730

**Comment by jgraham** [comment]
Well, it's the same test that fails each time…

**Comment by jdm** [comment]
@bors-servo: retry
* #7730
It's ok, it fails over and over in other PRs too.

**Comment by jgraham** [comment]
Ah, the other kind of OK

**Pull request merged**